addons/mail/static/src/web/messaging_menu/messaging_menu.xml

```diff
@@ -27,8 +27,8 @@
             </t>
             <div class="flex-grow-1"></div>
             <t t-if="!store.discuss.isActive or store.isSmall and store.discuss.activeTab !== 'mailbox'">
-                <t t-if="env.inDiscussApp and ['chat', 'channel'].includes(store.discuss.activeTab)">
-                    <div t-if="state.addingChat or state.addingChannel" class="p-4 border-bottom bg-light" t-ref="selector">
+                <t t-if="shouldAddChannelSelector">
+                    <div t-if="shouldDisplayChannelSelector" class="p-4 border-bottom bg-light" t-ref="selector">
```

Comment:
Both these conditions could be merged in a single one. And the getter could be renamed to `displayChannelSelector`.

And you forgot to handle `env.inDiscussApp` check.